Classes/Image Upload/ImageUploadViewController.swift

```diff
@@ -0,0 +1,214 @@
+//
+//  ImageUploadViewController.swift
+//  Freetime
+//
+//  Created by Sherlock, James on 30/09/2017.
+//  Copyright © 2017 Ryan Nystrom. All rights reserved.
+//
+
+import UIKit
+
+protocol ImageUploadDelegate: class {
+    func imageUploaded(link: String, altText: String)
+}
+
+class ImageUploadViewController: UIViewController, UITextFieldDelegate {
+
+    @IBOutlet private var previewImageView: UIImageView!
+    @IBOutlet private var titleTextField: UITextField!
+    @IBOutlet private var bodyTextField: UITextView!
+    
+    private var image: UIImage! // Set through the create function
+    private var username: String?
+    private weak var delegate: ImageUploadDelegate?
+    
+    private var compressionData: String?
+    private lazy var client = ImgurClient()
+    
+    private var titleText: String? {
+        guard let raw = titleTextField.text?.trimmingCharacters(in: .whitespacesAndNewlines) else { return nil }
+        if raw.isEmpty { return nil }
+        return raw
+    }
+    
+    private var descriptionText: String? {
+        let raw = bodyTextField.text.trimmingCharacters(in: .whitespacesAndNewlines)
+        if raw.isEmpty { return nil }
+        return raw
+    }
+    
+    class func create(_ image: UIImage, username: String?, delegate: ImageUploadDelegate) -> ImageUploadViewController? {
+        let storyboard = UIStoryboard(name: "ImageUpload", bundle: nil)
+        
+        guard let viewController = storyboard.instantiateInitialViewController() as? ImageUploadViewController else {
+            return nil
+        }
+        
+        viewController.image = image
+        viewController.username = username
+        viewController.delegate = delegate
+        
+        return viewController
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Set the preview image
+        previewImageView.image = image
+        
+        // Set title field delegate so return moves to next field
+        titleTextField.delegate = self
+        
+        // Add borders to imitate UITableView grouped design
```

Comment:
Why not a `UITableView` this time? 🤓

Reply:
Thought I fixed this 😂 It's meant to be a table view! Oops!

Reply:
Fixed!